lib/github_changelog_generator/generator/generator_processor.rb

```diff
@@ -178,8 +178,10 @@ def filter_merged_pull_requests(pull_requests)
         fetched_pr = closed_pull_requests.find do |fpr|
           fpr.number == pr.number
         end
-        pr[:merged_at] = fetched_pr[:merged_at]
-        closed_pull_requests.delete(fetched_pr)
+        unless fetched_pr.nil?
+          pr[:merged_at] = fetched_pr[:merged_at]
+          closed_pull_requests.delete(fetched_pr)
+        end
```

Comment:
Yes I was focused on making sure I checked for nil as that was the error, but now I look at it the other way is better, update coming.